src/lib/mirage_block_c.c

```diff
@@ -86,29 +87,31 @@ ocaml_mirage_block_stat(mirage_block_handle h, struct stat *out, int *err) {
 	int read_write = Int_val(Field(result, 0)) != 0;
 	unsigned int sector_size = (unsigned int)Int_val(Field(result, 1));
 	uint64_t size_sectors = (uint64_t)Int64_val(Field(result, 2));
+	int candelete = Bool_val(Field(result, 3));
 	if (Is_exception_result(result)){
 		*err = 1;
 	} else {
 		*err = 0;
-		bzero(out, sizeof(struct stat));
-		out->st_dev = 0;
-		out->st_ino = 0;
-		out->st_mode = S_IFREG | S_IROTH | S_IRGRP | S_IRUSR | (read_write?(S_IWOTH | S_IWGRP | S_IWUSR): 0);
-		out->st_nlink = 1;
-		out->st_uid = 0;
-		out->st_gid = 0;
-		out->st_rdev = 0;
-		out->st_size = (off_t)(sector_size * size_sectors);
-		out->st_blocks = (blkcnt_t)size_sectors;
-		out->st_blksize = (blksize_t)sector_size;
-		out->st_flags = 0;
-		out->st_gen = 0;
+		bzero(out->stat, sizeof(struct mirage_block_stat));
```

Comment:
Or another alternative would be to not embed the `stat` into the `mirage_block_stat` at all and simply add a new argument to this function for the mirage specific stuff.